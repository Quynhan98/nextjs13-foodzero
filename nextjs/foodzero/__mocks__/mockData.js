export const AVATAR_MOCK =
  'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf'

export const PRICE_LIST_MOCK = {
  name: 'Deep Sea Snow White Cod Fillet',
  price: 20,
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
}

export const CARD_BLOG = {
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=0a725961-6c86-4dcd-ab48-e5d7e2eb48fd',
  title: 'Fruit and vegetables and protection against diseases',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  category: 'Fashion',
  srcAvatar: AVATAR_MOCK,
  name: 'Julie Christie',
  date: '2022-12-06T23:29:35.000Z',
  comments: 2,
}

export const CARD_FEATURE = {
  title: 'Premium Quality',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FCircleIcon.webp?alt=media&token=1df373f3-27a3-45f6-82fa-d997c47c718f',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Neque congue arcu',
}

export const CARD_CATEGORY = {
  href: '/',
  category: 'Starters',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcategory.webp?alt=media&token=4e7dcf6c-2a66-4695-917d-af1d578d9a30',
}

export const USERS_MOCK = [
  {
    id: 'gff1',
    email: 'foodzero@gmail.com',
    password: '123456789',
    reservations: [],
  },
]

export const QUOTE_MOCK = [
  {
    id: '1',
    authorName: 'John Doe',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf',
  },
  {
    id: '2',
    authorName: 'Julie Christie',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet1.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Funnamed.webp?alt=media&token=f2607e25-62ae-4ed9-9f53-bb76e6f830fb',
  },
  {
    id: '3',
    authorName: 'Dianne Russell',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet2.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcute-avatar.webp?alt=media&token=9d70884c-e075-4f36-bf67-b196493badb8',
  },
]

export const OUR_MENU_MOCK = [
  {
    id: '1',
    name: 'Deep Sea Snow White Cod Fillet',
    price: 20,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit1.',
  },
  {
    id: '2',
    name: 'Steak With Rosemary Butter',
    price: 22,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
  },
  {
    id: '3',
    name: 'Cucumber Salad',
    price: 18,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit3.',
  },
  {
    id: '4',
    name: 'Natural Wine Pairing',
    price: 90,
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit4.',
  },
]

export const BLOG_SECTION_MOCK = [
  {
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=766a34ea-32e0-40e7-90e6-299f9af22eb1',
    title: 'Fruit and vegetables and protection against diseases',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit1.',
    category: 'Fashion',
    srcAvatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjulie.webp?alt=media&token=c429d5f2-6f41-4233-970b-b2739746f7f1',
    name: 'Julie Christies',
    date: '2021-10-17T23:29:35.000Z',
    comments: 2,
    id: '1',
  },
  {
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fasparagus.webp?alt=media&token=d56ab468-1479-4181-844d-f5b231742244',
    title: "Asparagus Spring Salad with Rocket, Goat's Cheese",
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
    category: 'Fashion',
    srcAvatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fdianne.webp?alt=media&token=ce3d7213-72ce-4caa-b43c-f6f07bb3b5b7',
    name: 'Dianne Russell',
    date: '2021-10-17T23:29:35.000Z',
    comments: 2,
    id: '2',
  },
]

export const MENU_MOCK = {
  id: '1',
  ourMenu: OUR_MENU_MOCK,
  starters: OUR_MENU_MOCK,
  mains: OUR_MENU_MOCK,
  pastriesAndDrinks: OUR_MENU_MOCK,
}
