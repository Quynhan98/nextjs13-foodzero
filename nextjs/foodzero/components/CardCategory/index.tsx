import { memo } from 'react'
import Image from 'next/image'
import Link from 'next/link'
import { Box, Flex, Text, useMediaQuery } from '@chakra-ui/react'

// Constants
import { BREAKPOINTS } from '@constants/index'

// Themes
import { rufina } from '@themes/index'

// Utils
import { imageDataUrl } from '@utils/convertToBase64'

interface CardCategoryProps {
  href: string
  category: string
  imageUrl: string
}

const CardCategory = ({ href, category, imageUrl }: CardCategoryProps) => {
  const [isMobile] = useMediaQuery(BREAKPOINTS.MEDIUM)

  return (
    <Box position="relative" maxW="508px" maxH="711px">
      <Flex
        position="absolute"
        zIndex="1"
        width="100%"
        justifyContent="space-between"
        padding={{ base: '25px 20px', '2xl': '50px 45px' }}
      >
        <Text
          size="extraLarge"
          variant="secondary"
          fontWeight="bold"
          fontFamily={rufina}
        >
          {category}
        </Text>
        <Link
          href={href}
          prefetch={false}
          style={{
            position: 'relative',
            width: isMobile ? '36px' : '46px',
            height: isMobile ? '20px' : '26px',
          }}
        >
          <Image
            fill
            src="/images/rightArrow.webp"
            alt="right arrow icon"
            sizes="(max-width: 768px) 36px, 20px
                (min-width: 1200px) 46px, 26px"
          />
        </Link>
      </Flex>
      <Image
        src={imageUrl}
        alt="background of the category card"
        width={508}
        height={711}
        placeholder="blur"
        blurDataURL={imageDataUrl(508, 711)}
      />
    </Box>
  )
}

export default memo(CardCategory)
