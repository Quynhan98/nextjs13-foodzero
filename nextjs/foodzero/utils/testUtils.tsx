import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
