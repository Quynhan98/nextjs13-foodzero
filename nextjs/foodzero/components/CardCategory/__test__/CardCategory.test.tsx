import { render, screen } from '@utils/testUtils'

// Mocks
import { CARD_CATEGORY } from '@mocks/mockData'

// Components
import CardCategory from '@components/CardCategory'

describe('CardCategory component', () => {
  const props = { ...CARD_CATEGORY }

  it('Renders CardCategory correctly', () => {
    const { container } = render(<CardCategory {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render data correctly', () => {
    render(<CardCategory {...props} />)

    const category = screen.getByText(props.category)
    const link = screen.getByRole('link')

    expect(category).toHaveTextContent(props.category)
    expect(link.getAttribute('href')).toEqual(props.href)
  })
})
