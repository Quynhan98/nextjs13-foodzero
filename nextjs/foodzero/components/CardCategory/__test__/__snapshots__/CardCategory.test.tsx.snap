// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCategory component Renders CardCategory correctly 1`] = `
<div>
  <div
    class="css-hfnpep"
  >
    <div
      class="css-ix7fvz"
    >
      <p
        class="chakra-text css-1x99fdl"
      >
        Starters
      </p>
      <a
        href="/"
        style="position: relative; width: 46px; height: 26px;"
      >
        <img
          alt="right arrow icon"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 36px, 20px (min-width: 1200px) 46px, 26px"
          src="/_next/image?url=%2Fimages%2FrightArrow.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2FrightArrow.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2FrightArrow.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </a>
    </div>
    <img
      alt="background of the category card"
      data-nimg="1"
      decoding="async"
      height="711"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252Fcategory.webp%3Falt%3Dmedia%26token%3D4e7dcf6c-2a66-4695-917d-af1d578d9a30&w=1080&q=75"
      srcset="/_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252Fcategory.webp%3Falt%3Dmedia%26token%3D4e7dcf6c-2a66-4695-917d-af1d578d9a30&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fimage-blog-155e2.appspot.com%2Fo%2Ffoodzero%252Fcategory.webp%3Falt%3Dmedia%26token%3D4e7dcf6c-2a66-4695-917d-af1d578d9a30&w=1080&q=75 2x"
      style="color: transparent;"
      width="508"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
